[Inquiry] Is valkey-glide meant to work on go vendor mode?

### Inquiry

We typically vendor modules (go mod vendor) and commit into VCS. 

After this https://github.com/valkey-io/valkey-glide/pull/5071 was merged, lib.h and other dependencies are in this repo's [.gitignore](https://github.com/valkey-io/valkey-glide/blob/main/go/.gitignore) file. 

Due to this, when I vendor valkey-glide into my project, it won't commit these files into our VCS.

Is valkey-glide meant to work on projects that commit the vendor/? 

Thanks!


### Language

Go

### Language Version

1.26.2

### Engine Version

_No response_

### Operating System

_No response_

### Additional Technical Information

_No response_

Have created a [sample project](https://github.com/severedsea/valkey-glide-go-vendor-ignore) to help in the investigation. 

Steps to reproduce
1. Checkout repository
2. Do not call `go mod vendor`
3. Run `make up` to start valkey
4. Execute `go run main.go`

Expected results
```
vendor/github.com/valkey-io/valkey-glide/go/v2/config/ClientSideCache.go:9:2: cannot find module providing package github.com/valkey-io/valkey-glide/go/v2/internal/protobuf: import lookup disabled by -mod=vendor
	(Go version in go.mod is at least 1.14 and vendor directory exists.)
```

Thanks for your help!

---

As a workaround, our team has created a new make target (`make vendor`) to execute `go mod vendor` and added a hook to delete `valkey-glide`'s .gitignore, to ensure that it won't accidentally get deleted. 

```
rm -f vendor/github.com/valkey-io/valkey-glide/go/v2/.gitignore
```

--- a/.github/workflows/go-cd.yml
+++ b/.github/workflows/go-cd.yml
@@ -232,6 +232,13 @@ jobs:
                   RELEASE_VERSION=${{ needs.validate-release-version.outputs.RELEASE_VERSION }}
                   git config user.name github-actions
                   git config user.email github-actions@github.com
+                  # Remove go/.gitignore from the released module. 'go mod vendor'
+                  # copies this file into the consumer's vendor/ tree, where its
+                  # rules (*.pb.go, rustbin/, lib.h) cause git to silently drop the
+                  # generated artifacts we force-add below when the consumer commits
+                  # vendor/. This tag commit is never merged back to main, so main
+                  # keeps its .gitignore for development. See issue #6437.
+                  rm -f $GITHUB_WORKSPACE/go/.gitignore
                   git add -f go
                   git status
                   git commit -m "Automated commit from GitHub Actions"

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@
 
 * Core: Enforce the RESP3 parser recursion-depth limit for all aggregate types (map, set, push, attribute), not just arrays. A malicious or compromised server could previously send deeply nested `%`/`~`/`>`/`|` payloads that consumed one native stack frame per level and crashed the host application via stack exhaustion (DoS); such payloads now surface a graceful parse error. ([#6477](https://github.com/valkey-io/valkey-glide/pull/6477))
 * Core: Update `anyhow` to 1.0.103 to fix RUSTSEC-2026-0190, an unsoundness advisory in `anyhow::Error::downcast_mut()` that can trigger undefined behavior ([#6364](https://github.com/valkey-io/valkey-glide/pull/6364))
+* Go: Remove `.gitignore` from the released module so consumers who commit `vendor/` keep the generated artifacts (`internal/protobuf/*.pb.go`, `rustbin/**`, `lib.h`) ([#6441](https://github.com/valkey-io/valkey-glide/pull/6441))
 
 ### Changes
 

